tchannel/tornado/connection.py

```diff
@@ -444,6 +442,7 @@ def _extract_handshake_headers(self, message):
         self.remote_host_port = int(self.remote_host_port)
         self.remote_process_name = message.process_name
         self.requested_version = message.version
+        self._handshake_performed = True
```

Comment:
this isn't a handshake header.  We should set this explicitly in initiate_handshake and expect_handshake